wwwbase/editare-sursa.php

```diff
@@ -21,22 +21,52 @@
   $src->canDistribute = util_getRequestParameterWithDefault("canDistribute", 0);
   $src->defCount = util_getRequestIntParameter("defCount");
 
+  $existing_curators = Curator::getCuratorsForSource($sourceId);
+  $curators = array_unique(util_getRequestIntArrayParameter('curators'));
+
   if (validate($src)) {
     // For new sources, set displayOrder to the highest available + 1
     if (!$sourceId) {
       $src->displayOrder = Model::factory('Source')->count() + 1;
     }
     $src->updatePercentComplete();
-    $src->save();
-    FlashMessage::add('Modificările au fost salvate', 'info');
+    try {
+      ORM::get_db()->beginTransaction();
+      $src->save();
+      // Delete curators no longer assigned
+      $existing_ids = array();
+      /** @var User $user */
+      foreach($existing_curators as $user) {
+        $existing_ids[] = $user->id();
+        if(!in_array($user->id(), $curators)) {
+          Model::factory('Curator')->where(array('sourceId' => $sourceId, 'userId' => $user->id()))->delete_many();
+        }
+      }
+      // Attach new curators
+      foreach($curators as $userId) {
+        if(!in_array($userId, $existing_ids)) {
+          $curator = Model::factory('Curator')->create();
+          $curator->sourceId = $sourceId;
+          $curator->userId = $userId;
+          $curator->save();
+        }
+      }
```

Comment:
E frumos că ai făcut diff ca să modifici cât mai puțin. Ar fi fost ok și să ștergi toți curatorii și să-i salvezi pe toți cei noi. Facem asta la toate asocierile many-to-many, de ex. cuvinte-definiții.